rhel/matcher.go

```diff
@@ -1,37 +1,61 @@
 package rhel
 
 import (
+	"context"
+	"net/http"
+	"os"
+
 	version "github.com/knqyf263/go-rpm-version"
 
 	"github.com/quay/claircore"
 	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/rhel/repo2cpe"
 )
 
+// DefaultRepo2CPEMappingURL is default URL with a mapping file provided by Red Hat
+const DefaultRepo2CPEMappingURL = "https://www.redhat.com/security/data/metrics/repository-to-cpe.json"
+
 // Matcher implements driver.Matcher.
-type Matcher struct{}
+type Matcher struct {
+	mapping *repo2cpe.RepoCPEMapping
+}
 
 var _ driver.Matcher = (*Matcher)(nil)
 
+func NewMatcher() *Matcher {
```

Comment:
When will this constructor return? 

Will it return only after the content mapping file is loaded into memory?
Will it return before the mapping file is loaded into memory?

If it returns before the mapping file is loaded into memory, calls to `m.mapping.RepositoryToCPE` should *not* error. If the mapping file is not in memory, the matching process should gracefully return no matches. 

Can you confirm we come up with a good error scenario for this?

Reply:
`m.mapping.RepositoryToCPE` updates a mapping if it is not loaded yet. The constructor doesn't load any data.

Reply:
Yes, but what if this "Update" fails because the mapping file cannot be reached? The matching process should not fail. You are now going to an external source, and if thats not available, we shouldnt error the matching process, 

Is that handles gracefully?